explain rationale for new application protocols to support TLS 1.2 (#464)
yaronf/I-D: BCP195bis/draft-ietf-uta-rfc7525bis.md

               
  Rationale: TLS 1.3 is a major overhaul to the protocol and resolves many of the security issues with TLS 1.2. To the extent that an implementation supports TLS 1.2 (even if it defaults to TLS 1.3), it MUST follow the recommendations regarding TLS 1.2 specified in this document.

* New transport protocols that integrate the TLS/DTLS handshake protocol and/or record layer MUST use only TLS/DTLS 1.3 (for instance, QUIC {{RFC9001}}) took this approach) and new application protocols that employ TLS/DTLS for channel or session encryption MUST use TLS/DTLS 1.2 and/or 1.3.  Implementations would then adhere to the TLS/DTLS version used by the relevant protocol.

>   Rationale: Although secure deployment of TLS 1.3 is easier and less error-prone than TLS 1.2 because of the need to follow the recommendations provided in this document for secure deployment of TLS 1.2, development of new transport protocols is significantly more complex than simply re-using the existing support for TLS/DTLS in underlying libraries or operating systems as is usually done in application protocols.

Reviewer: I think the previous version was clearer.

```suggestion
  Rationale: Secure deployment of TLS 1.3 is significantly easier and less error-prone than secure deployment of TLS 1.2. When designing a new secure transport protocols such as QUIC, there is no reason to support TLS 1.2. By contrast, new application protocols that re-use TLS MAY support both TLS 1.3 and TLS 1.2 in order to take advantage of underlying library or operating system support for both versions.
```

* New protocol designs that embed TLS mechanisms SHOULD use only TLS 1.3 and SHOULD NOT use TLS 1.2; for instance, QUIC {{RFC9001}}) took this approach. As a result, implementations of such newly-developed protocols SHOULD support TLS 1.3 only with no negotiation of earlier versions.

>   Rationale: Secure deployment of TLS 1.3 is significantly easier and less error-prone than secure deployment of TLS 1.2.  This advice applies especially to new security-focused protocols such as QUIC, where there is likely no reason to support TLS 1.2.  By contrast, new application protocols that re-use TLS might find it helpful to support both TLS 1.3 and TLS 1.2 in order to take advantage of underlying library or operating support for both versions, and thus avoid the significant effort involved in building a security-focused protocol such as QUIC.

Reviewer: ```suggestion
  Rationale: Secure deployment of TLS 1.3 is significantly easier and less error-prone than secure deployment of TLS 1.2.  This advice applies especially to new security-focused protocols such as QUIC, where there is likely no reason to support TLS 1.2.  By contrast, new application protocols that re-use TLS might find it helpful to support both TLS 1.3 and TLS 1.2 in order to take advantage of underlying library or operating support for both versions, and thus avoid the significant effort involved in building a secure transport protocol such as QUIC.
```

>   Rationale: Secure deployment of TLS 1.3 is significantly easier and less error-prone than secure deployment of TLS 1.2.  This advice applies especially to new secure transport protocols such as QUIC, where there is likely no reason to support TLS 1.2.  By contrast, new application protocols that re-use TLS might find it helpful to support both TLS 1.3 and TLS 1.2 in order to take advantage of underlying library or operating system support for both versions, and thus avoid the significant effort involved in designing a new secure transport protocol such as QUIC.

Reviewer: ```suggestion
  Rationale: Secure deployment of TLS 1.3 is significantly easier and less error-prone than secure deployment of TLS 1.2.  This advice applies especially to new secure transport protocols such as QUIC, where there is likely no reason to support TLS 1.2.  By contrast, new application protocols that re-use TLS MAY support both TLS 1.3 and TLS 1.2 in order to take advantage of underlying library or operating system support for both versions, and thus avoid the significant effort involved in designing a new secure transport protocol such as QUIC.
```

Reviewer: Not sure about:
_"[...] and thus avoid the significant effort involved in designing a new secure transport protocol such as QUIC."_
is a super compelling reason.

A more practical reason might be that you and I create a new protocol on top of HTTP/2 and we want to have the wider possible user base.  There is a huge amount of library code out there that does HTTP/2 over TLS 1.2 which we'd want to (and can) capture.
